### DISCUSSION TITLE
Build system guide?

### DISCUSSION THREAD

**Discussion by maxxer** [oerdnj/deb.sury.org#1848] [General]
Hi.
I was wondering if the packages build system is open source as well. I mean, are all the tools and scripts used to build new releases for all the supported distros available?

I'd like to do it for another project, and I was wondering if I can avoid starting from scratch.

Thank you

**Comment by rfay** [comment]
This is super important IMO, and not just for other users, but for the sustainability of this project.
* https://github.com/oerdnj/deb.sury.org/issues/1436

**Reply by oerdnj** [reply]
The repositories listed in #1436 are still up-to-date. The rest is you need to glue it together with dput and PPAs for Ubuntu and jenkins-job-builder and https://jenkins-debian-glue.org + pbuilder for Debian. There are some unimportant scripts for pushing the local builder apt repository to the remote mirror, but I can't publish those as they include the credentials for the CDN.

There's not much else - I update the source package in git, refresh the patches (if needed), push the git repository, run `ubuntu-ppa-build -p php` (the one from the packaging-tools) for Ubuntu packages, and launch the jobs in Jenkins for Debian. That's it.

**Reply by oerdnj** [reply]
Basically, it's pretty much standard Debian packaging stuff. I've just tweaked some bits here and there, so it automates the backporting for multiple distributions. Some pieces are still missing, f.e. I could write automation to launch the right Jenkins jobs on a tag or something like this, but so far it's easier to just click the mouse couple of times.

**Reply by oerdnj** [reply]
Sometimes I wish I had some trade secret, that I could sell for big bucks and never work again :-), but it's all just built around the trust that I do a solid job. In the end, it's all for the better like this.

**Reply by maxxer** [reply]
So the very same source is used for all distros? Doesn't it happen to have patches that won't apply to a certain OS, does it?

**Reply by oerdnj** [reply]
> So the very same source is used for all distros?

Yep, I keep the packaging at the lowest common denominator...
